### PR TITLE
Fixed #25896 -- Fixed state bug in SeparateDatabaseAndState.database_backwards().

### PR DESCRIPTION
**Pull request by amosonn**
Test breaking with previous code, and fix which makes test pass.

### CODE REVIEW

File: tests/migrations/test_operations.py [hunk 1]
```diff
@@ -1835,6 +1835,80 @@ def test_separate_database_and_state(self):
         self.assertEqual(definition[1], [])
         self.assertEqual(sorted(definition[2]), ["database_operations", "state_operations"])
 
+    @unittest.skipIf(sqlparse is None and connection.features.requires_sqlparse_for_splitting, "Missing sqlparse")
```
Comment (timgraham) [inline tests/migrations/test_operations.py]: There's no RunSQL in this test, so I don't think sqlparse is required.

File: tests/migrations/test_operations.py [hunk 2]
```diff
@@ -1835,6 +1835,80 @@ def test_separate_database_and_state(self):
...
+    def test_separate_database_and_state2(self):
+        """
+        Tests the SeparateDatabaseAndState operation.
```
Comment (timgraham) [inline tests/migrations/test_operations.py]: A description about how it differs from the other test would be nice.

File: tests/migrations/test_operations.py [hunk 3]
```diff
@@ -1835,6 +1835,80 @@ def test_separate_database_and_state(self):
...
+        ]
+        operation = migrations.SeparateDatabaseAndState(
+            state_operations=state_operations,
+            database_operations=database_operations,
+        )
+        # Test the state alteration
+        new_state = project_state.clone()
+        operation.state_forwards(app_label, new_state)
+
+        def assert_models_and_tables(after_db):
```
Comment (timgraham) [inline tests/migrations/test_operations.py]: Usually assertions like this are camel cased.

**Comment by timgraham** [comment]
Please add a mention in the 1.8.8 release notes. For future reference, please send pull requests against master and the comitter will take care of backporting as necessary.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by amosonn** [comment]
I did all of them, and rebased to master to a new branch. Do I close this request and open a new one?

**Comment by timgraham** [comment]
Yes, you can't edit the branch a pull request is targeted at.

**Pull request closed without merging**